### PR TITLE
NIP-01: clarify single filter match

### PR DESCRIPTION
**Pull request by patrickReiis**
I didn't understand it first time when I read the current explanation, I think with this merge request it'll be easier for people with no nostr experience to understand it first time.

### CODE REVIEW

File: 01.md [hunk 1]
```diff
@@ -139,7 +139,7 @@ The `ids`, `authors`, `#e` and `#p` filter lists MUST contain exact 64-character
 
 The `since` and `until` properties can be used to specify the time range of events returned in the subscription. If a filter includes the `since` property, events with `created_at` greater than or equal to `since` are considered to match the filter. The `until` property is similar except that `created_at` must be less than or equal to `until`. In short, an event matches a filter if `since <= created_at <= until` holds.
 
-All conditions of a filter that are specified must match for an event for it to pass the filter, i.e., multiple conditions are interpreted as `&&` conditions.
+All different fields of a filter, if present, must match for an event for it to pass the filter. For example, if a filter has `{"kinds": [1, 3], "authors": [<pubkey>]}` it'll be interpreted as `(kind === 1 || kind === 3) && author === <pubkey>`.
```
Comment (AsaiToshiya) [inline 01.md:142]:
```suggestion
All different fields of a filter, if present, must match for an event for it to pass the filter. For example, if a filter has `{"kinds": [1, 3], "authors": [<pubkey>]}` it'll be interpreted as `(kind == 1 || kind == 3) && author == <pubkey>`.
```

I think the equality operator is `==` in many languages.

**Review by AsaiToshiya — APPROVED** [review]

File: 01.md (hunk 1 above)
Comment (patrickReiis) [reply 01.md:142]: Sorry, I mostly Deno. For me it doesn't matter, committing your suggestion.

**Comment by patrickReiis** [comment]
What do you think? @staab

**Comment by alexgleason** [comment]
The original text is concise. An example helps, but I'm not sure it belongs there. 

Maybe it would make more sense to put a bulleted list of some examples at the end of the section, similar to the relay "OK" message examples.

**Comment by alexgleason** [comment]
I was thinking the same @Semisol

**Comment by patrickReiis** [comment]
How would tests look like? Using this merge request as an example.

**Comment by bezysoftware** [comment]
I would loooove to see gherkin tests for NIPs (either along each NIP or maybe in a separate repo?) This is a sample test from nostream:
```gherkin
Feature: NIP-01
  Scenario: Alice gets an event by ID
    Given someone called Alice
    And someone called Bob
    When Bob sends a text_note event with content "why is nobody talking about this?"
    And Alice subscribes to last event from Bob
    Then Alice receives a text_note event from Bob with content "why is nobody talking about this?"
```

I'm sure this could be adapted to cover the filter matching

**Comment by bezysoftware** [comment]
I wrote a few NIP-01 gherkin tests for my Netstr project, check them out: https://github.com/bezysoftware/netstr/blob/main/test/Netstr.Tests/NIPs/01.feature

**Pull request closed without merging**